applications/MeshingApplication/custom_utilities/tetrahedra10_mesh_converter_utility.h

```diff
@@ -0,0 +1,356 @@
+	int edge_ids[6];
+	std::vector<int> node_ids;
+
+	for (ElementsArrayType::iterator it = it_begin; it != it_end; ++it)
+	{
+        GlobalPointersVector< Element >& rChildElements = it->GetValue(NEIGHBOUR_ELEMENTS);
+        rChildElements.resize(0);
+
+        CalculateEdges(it->GetGeometry(), Coord, edge_ids, node_ids);
+
+        // Generate the new Tetrahedra3D10 element
+        Tetrahedra3D10<Node<3>> geom = GenerateTetrahedra(this_model_part, node_ids);
+        const Element& rElem = KratosComponents<Element>::Get("Element3D10N");
+        Element::Pointer p_element;
+        p_element = rElem.Create(it->Id(), geom, it->pGetProperties());
+        p_element->Initialize(rCurrentProcessInfo);
+        p_element->InitializeSolutionStep(rCurrentProcessInfo);
+        p_element->FinalizeSolutionStep(rCurrentProcessInfo);
+
+        // Setting the internal variables in the "child" elem (the element replacing the old one)
+        if (interpolate_internal_variables == true)
+        {
+            //This method only copies the current information to the new element
+            InterpolateInteralVariables(0, *it.base(), p_element, rCurrentProcessInfo);
+        }
+
+        // Transfer elemental variables to new element
+        p_element->Data() = it->Data();
+        p_element->GetValue(SPLIT_ELEMENT) = false;
+        NewElements.push_back(p_element);
+
+        rChildElements.push_back( Element::WeakPointer(p_element) );
+    }
+
+    // Now replace the elements in SubModelParts
+    if ( NewElements.size() > 0 ) ReplaceElementsInSubModelPart(this_model_part);
+}
+
+    /**
+    * It replaces the old Tetrahedra3D4 elements in its corresponding submodelpart by its new Tetrahedra3D10
+    * @param this_model_part: The modelpart or submodelpart to replace the elements
+    */
+    void ReplaceElementsInSubModelPart(ModelPart& this_model_part) {
+        for(Element::Pointer& p_element : this_model_part.ElementsArray()){
+                if( p_element->GetValue(SPLIT_ELEMENT) )
+                {
+                    GlobalPointersVector< Element >& children = p_element->GetValue(NEIGHBOUR_ELEMENTS);
+                    p_element = children[0].shared_from_this();
+                } 
+        }
+
+        //Recursively for all subModelParts
+        for (ModelPart::SubModelPartIterator iSubModelPart = this_model_part.SubModelPartsBegin();
+                iSubModelPart != this_model_part.SubModelPartsEnd(); iSubModelPart++)
+        {
+            ReplaceElementsInSubModelPart(*iSubModelPart);
+        }
+    }
+
+    /**
+    * Remove the old Trangle3D3 conditions and replace them by the new Trangle3D6 ones
+    * @param Coord: The coordinates of the nodes of the geometry
+    * @return this_model_part: The model part of the model (it is the input too)
+    */
+
+    void EraseOldConditionsAndCreateNew(
+            ModelPart& this_model_part,
+            const compressed_matrix<int>& Coord
```

Comment:
Can't we use an std::vector instead?

Reply:
This function is being called from the base class, where a compressed_matrix is used. We could use std::vector but we would have to re-implement the base class and all the subclasses (not very worthy I think).

Reply:
OK. Yes, we can eventually change it in a different PR. The point is to, step by step, get rid of the Boost dependency.